addons/web/static/src/views/fields/properties/property_tags.js

```diff
@@ -0,0 +1,237 @@
+        return JSON.parse(JSON.stringify(this.props.tags || []));
+    }
+
+    /**
+     * Options available in the autocomplete component.
+     */
+    get autocompleteActions() {
+        return [{
+            options: (request) => {
+                const tagsFiltered = this.props.tags.filter(tag =>
+                    (!this.props.value || this.props.value.indexOf(tag[0]) < 0)
+                    && (
+                        !request || !request.length
+                        || tag[1].toLocaleLowerCase().indexOf(request.toLocaleLowerCase()) >= 0
+                    )
+                );
+                if (!tagsFiltered || !tagsFiltered.length) {
+                    // no result, ask the user if he want to create a new tag
+                    if (!request || !request.length) {
+                        return [{
+                            value: null,
+                            label: _lt('Start typing...'),
+                            classList: 'fst-italic',
+                        }];
+                    } else if (!this.props.canChangeTags) {
+                        return [{
+                            value: null,
+                            label: _lt('No result'),
+                            classList: 'fst-italic',
+                        }];
+                    }
+
+                    return [{
+                        value: {toCreate: true, value: request},
+                        label: sprintf(_lt('Create "%s"'), request),
+                        classList: 'o_field_property_dropdown_add',
+                    }];
+                }
+                return tagsFiltered.map((tag) => {
+                    return {
+                        value: tag[0],
+                        label: tag[1],
+                    };
+                });
+            },
+        }];
+    }
+
+    /**
+     * Add one value in the current tag list values.
+     */
+    onOptionSelected(tagValue) {
+        if (!tagValue) {
+            // clicked on "Start typing..."
+            return;
+        }
+
+        if (tagValue.toCreate) {
+            this.onNewTag(tagValue.value);
+        } else {
+            const currentValue = this.currentValue;
+            const newValue = [...currentValue, tagValue];
+            this.props.onValueChange(newValue);
+        }
+    }
+
+    /**
+     * Ask to create a new tag that will be added in
+     * the definition and automatically selected.
+     */
+    onNewTag(newLabel) {
+        if (!newLabel || !newLabel.length) {
+            return;
+        }
+
+        const newValue = this.labelToValue(newLabel);
+
+        const existingTags = this.props.tags.find(tag => tag[0] === newValue);
+        if (existingTags) {
+            this.notification.add(
+                _lt('This tag is already available'),
+                { type: 'warning' },
+            );
+            return;
+        }
+
+        const tagColor = parseInt(Math.random() * ColorList.COLORS.length);
```

Comment:
Should we rather cycle through available colors?

I tested and got a lot of white tags, which was a bit confusing:

![image](https://user-images.githubusercontent.com/41282536/184153319-fd2c3e64-4b94-48c8-8210-9f41f706e9c1.png)